tools/src/main/java/com/msd/gin/halyard/tools/HttpSparqlHandler.java

```diff
@@ -187,9 +188,19 @@ private SparqlQuery retrieveQuery(HttpExchange exchange) throws IOException, Ill
         }
         if ("GET".equalsIgnoreCase(requestMethod)) {
             // Retrieve from the request URI parameter query and optional parameters defaultGraphs and namedGraphs
-            String requestQuery = exchange.getRequestURI().getQuery();
-            if (requestQuery != null) {
-                StringTokenizer stk = new StringTokenizer(requestQuery, AND_DELIMITER);
+            // Cannot apply directly exchange.getRequestURI().getQuery() since getQuery() method
+            // automatically decodes query (requestQuery must remain unencoded due to parsing by '&' delimiter)
+            String requestQueryRaw = exchange.getRequestURI().toString();
```

Comment:
wouldn't exchange.getRequestURI().getRawQuery() work here?

Reply:
It definitely would :)